### PR TITLE
CA-205 | Add JAX-RS collection annotations for Java

### PR DESCRIPTION
**Pull request by sriram-privado**
Add missing JAX-RS collection annotations.

### CODE REVIEW

**Review by ankit-privado — APPROVED** [review]
LGTM

File: rules/collections/annotations/java.yaml [hunk 1]
```diff
@@ -34,3 +34,9 @@ collections:
     patterns:
       - "GET|POST"
     tags:
+
+  - id: Collections.Annotation.JAX-RS
+    name: JAX-RS Annotation
+    patterns:
+      - "(?i)(javax.ws.rs).(GET|PUT|POST|DELETE|HEAD|OPTIONS).*"
```
Comment (HiteshMah-Jan) [inline rules/collections/annotations/java.yaml:41]: please check if we match method fullnames for collection points. AFAIK its just the annotation name.

Comment (sriram-privado) [reply rules/collections/annotations/java.yaml:41]: I will check this.

Comment (sriram-privado) [reply rules/collections/annotations/java.yaml:41]: @HiteshMah-Jan we are only checking the 'name' and not the method fullname. You are right. We are using `cpg.annotation.name` to figure out matches.

Comment (sriram-privado) [reply rules/collections/annotations/java.yaml:41]: This rule would not work then. Can I just change it to `".*(GET|PUT|POST|DELETE|HEAD|OPTIONS).*"`, this way it'll match case-sensitive names for these ? @HiteshMah-Jan

**Pull request merged**